pkg/reconciler/identity/resources/service_account.go

```diff
@@ -34,11 +34,11 @@ func GenerateServiceAccountName(gServiceAccount, clusterName string) string {
 }
 
 // MakeServiceAccount creates a K8s ServiceAccount object for the Namespace.
-func MakeServiceAccount(namespace string, gServiceAccount, clusterName string) *corev1.ServiceAccount {
+func MakeServiceAccount(namespace string, kServiceAccount, gServiceAccount, clusterName string) *corev1.ServiceAccount {
```

Comment:
Four strings in a row makes me nervous, can we use an arguments struct instead?

Or reuse the struct I recommend in the reconciler.